File: 0232.implement_queue_using_stacks/solution.java
class MyQueue {
    private Stack<Integer> stack;
    private Stack<Integer> stackTemp;

    public MyQueue() {
        stack = new Stack<>();
        stackTemp = new Stack<>();
    }

    public void push(int x) {
        stack.push(x);
    }

    public int pop() {
        while (!this.empty()){
            stackTemp.push(stack.pop());
        }
        int temp = stackTemp.pop();
        while (!stackTemp.isEmpty()){
            stack.push(stackTemp.pop());
        }
        return temp;
    }

    public int peek() {
        while (!this.empty()){
            stackTemp.push(stack.pop());
        }
        int temp = stackTemp.peek();
        while (!stackTemp.isEmpty()){
            stack.push(stackTemp.pop());
        }
        return temp;
    }

    public boolean empty() {
        return stack.isEmpty();
    }
}

File: 0232.implement_queue_using_stacks/solution.py
class MyQueue:
    def __init__(self):
        self.queue = []

    def push(self, x):
        self.queue.append(x)

    def pop(self):
        if not self.empty():
            return self.queue.pop(0)
        return -1

    def peek(self):
        if not self.empty():
            return self.queue[0]
        return -1

    def empty(self):
        return not bool(len(self.queue))
